src/backend/shared/aws.lambda-shared.ts

```diff
@@ -30,22 +37,24 @@ export function s3(): AWS.S3 {
 /**
  * Checks whether an object exists in S3 at the provided bucket and key.
  */
-export function s3ObjectExists(bucket: string, key: string): Promise<boolean> {
-  return s3()
-    .headObject({
-      Bucket: bucket,
-      Key: key,
-    })
-    .promise()
-    .then(
-      () => true,
-      (cause) => {
-        if (cause.code === 'NotFound') {
-          return false;
-        }
-        return Promise.reject(cause);
-      }
+export async function s3ObjectExists(
+  bucket: string,
+  key: string
+): Promise<boolean> {
+  try {
+    await S3_CLIENT.send(
+      new HeadObjectCommand({
+        Bucket: bucket,
+        Key: key,
+      })
     );
+    return true;
+  } catch (cause: any) {
+    if (cause.code === 'NotFound') {
+      return false;
+    }
```

Comment:
This might still work, but the docs [1] indicate we should use `cause.name` and probably also `instanceof NotFound`
```suggestion
    if (cause instanceof NotFound || cause.name === 'NotFound') {
      return false;
    }
```


[1] https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/